extensions/interactions/ImageClickInput/directives/ImageClickInput.js

```diff
@@ -47,8 +51,28 @@ oppia.directive('oppiaInteractiveImageClickInput', [
               '/' + encodeURIComponent($scope.filepath)) : null);
           $scope.mouseX = 0;
           $scope.mouseY = 0;
+          $scope.interactionIsActive = !$scope.getLastAnswer();
           $scope.currentlyHoveredRegions = [];
           $scope.allRegions = imageAndRegions.labeledRegions;
+          $scope.updateCurrentlyHoveredRegions = function() {
+            for (var i = 0; i < imageAndRegions.labeledRegions.length; i++) {
+              var labeledRegion = imageAndRegions.labeledRegions[i];
+              var regionArea = labeledRegion.region.area;
+              if (regionArea[0][0] <= $scope.mouseX &&
+                  $scope.mouseX <= regionArea[1][0] &&
+                  regionArea[0][1] <= $scope.mouseY &&
+                  $scope.mouseY <= regionArea[1][1]) {
+                $scope.currentlyHoveredRegions.push(labeledRegion.label);
+              }
+            }
+          };
+          if ($scope.getLastAnswer()) {
+            /*The following lines permanently highlight the learner's previous
+              answer. */
+            $scope.mouseX = $scope.getLastAnswer().clickPosition[0];
```

Comment:
Don't we also need to do this if $scope.getLastAnswer() is null at the start, and then the learner subsequently clicks the right answer that then becomes the last one for the card?

Reply:
You mean when continue button appears after he has clicked right answer?
I am setting interactionIsActive as false when EVENT_NEW_CARD_AVAILABLE and am disabling image highlighting on mouse move over image based on that, so his last answer does get permanently highlighted.

Reply:
Ah, OK. Maybe add an explanation in the code to say that this only needs to be run for previous cards, since [... reason you gave above ...].

Reply:
Added.